Pod/Classes/Logging/Log.swift

```diff
@@ -61,59 +59,78 @@ open class Log {
     open static var printColoredText: Bool = false
 
-    
+    /// If true, prints emojis to signify log type, defaults to off
+    open static var useEmoji: Bool = false
+
     // MARK: Private
-    
-    fileprivate struct Cmd {
-        static let esc = "\u{001b}["
-        static let resetFG = esc + "fg;"
-        static let resetBG = esc + "bg;"
-        static let reset = esc + ";"
-    }
-    
-    fileprivate struct Color {
-        static let red = Cmd.esc + "fg240,0,0;"
-        static let yellow = Cmd.esc + "fg200,200,0;"
-        static let green = Cmd.esc + "fg0,150,0;"
+
+    /// Date formatter for log
+    fileprivate static let dateformatter: DateFormatter = {
+        let df = DateFormatter()
+        df.dateFormat = "Y-MM-dd H:m:ss.SSSS"
+        return df
+    }()
+
+    /// Generic log method
+    fileprivate static func log<T>(_ object: @autoclosure () -> T, level: Log.Level, _ fileName: String, _ functionName: String, _ line: Int) {
+        if logLevel.rawValue <= level.rawValue {
+            let date = Log.dateformatter.string(from: Date())
+            let components: [String] = fileName.components(separatedBy: "/")
+            let objectName = components.last ?? "Unknown Object"
+            let levelString = Log.useEmoji ? level.emoji() : "|" + level.name().uppercased() + "|"
+            print("\(levelString)\(date) \(objectName) \(functionName) line \(line):\n\(object())\n")
+        }
     }
-    
-    fileprivate static func log<T>(_ object: T, level: Level, color: String, _ fileName: String, _ functionName: String, _ line: Int) {
+
+    /// Generic log method
+    fileprivate static func log<T>(_ object: @autoclosure () -> T, level: Log.Level, _ fileName: String, _ functionName: String, _ line: Int) {
         if logLevel.rawValue <= level.rawValue {
-            let logColor = Log.printColoredText ? color : ""
-            let components:[String] = fileName.components(separatedBy: "/")
+            let date = Log.dateformatter.string(from: Date())
+            let components: [String] = fileName.components(separatedBy: "/")
             let objectName = components.last ?? "Unknown Object"
-            print( logColor + "\(Date())|\(level.name().uppercased())|\(objectName) \(functionName) line \(line): " + Cmd.reset + "\(object)")
+            let levelString = Log.useEmoji ? level.emoji() : "|" + level.name().uppercased() + "|"
+            print("\(levelString)\(date) \(objectName) \(functionName) line \(line):\n\(object())\n")
         }
     }
-    
+
     // MARK: Log Methods
     
-    static open func error<T>(_ object: T, _ fileName: String = #file, _ functionName: String = #function, _ line: Int = #line) {
-        log(object, level:.error, color: Color.red, fileName, functionName, line)
+    open static func error<T>(_ object: @autoclosure () -> T, _ fileName: String = #file, _ functionName: String = #function, _ line: Int = #line) {
```

Comment:
+1 for making them `public` instead of `open`, my guess is they were made extra permissive by the swift migrator